frontend/src/App.jsx

```
@@ -14,48 +14,63 @@ import { API_BASE_URL } from './config.js';
 
 function App() {
   const [user, setUser] = useState(null);
-  const [isLoading, setIsLoading] = useState(true);
+  const [flightData, setFlightData] = useState(initialFlightData);
+  const { isLoading, setIsLoading } = useLoading();
+  const [error, setError] = useState(null);
 
-  const updateUser = (newUser) => {
-    setUser(newUser);
+  const fetchUserData = async () => {
+    setIsLoading(true);
+    try {
+      const response = await fetch(`${API_BASE_URL}/users/data`, {
+        credentials: 'include',
+      });
+      if (response.ok) {
+        const userData = await response.json();
+        updateUser(userData);
+      } else {
+        setUser(null);
```

Shall we throw an exception here to use same logic as in catch block?